clippy_config/src/types.rs

```diff
@@ -23,12 +29,34 @@ impl DisallowedPath {
         path
     }
 
+    pub fn diag_amendment(&self, span: Span) -> impl FnOnce(&mut Diag<'_, ()>) + use<'_> {
+        move |diag| {
+            if let Some(replacement) = self.replacement() {
+                diag.span_suggestion(
+                    span,
+                    self.reason().map_or_else(|| String::from("use"), ToOwned::to_owned),
+                    replacement,
+                    Applicability::MachineApplicable,
```

Comment:
I'm not fully sure this is the right choice, I would be curious first to hear what others have to say. It's not guaranteed to be machine applicable but it is something the user intends to always be replaced and it's also at the user's discretion to provide something valid, and it could get annoying. The majority of the improvement does come from the fact that it can be applied automatically after all.

Reply:
> ...it's also at the user's discretion to provide something valid...The majority of the improvement does come from the fact that it can be applied automatically after all.

This was precisely my reasoning.

Reply:
It's sound. I doubt others will disagree but we'll see regardless